proxy/grpcproxy/watch.go

```diff
@@ -234,8 +237,35 @@ func (wps *watchProxyStream) sendLoop() {
 			if !ok {
 				return
 			}
-			if err := wps.stream.Send(wresp); err != nil {
-				return
+			// The grpc package's defaultServerMaxSendMessageSize, defaultServerMaxReceiveMessageSize
+			// and the maxSendMesgSize are set to lower numbers in order to demonstrate
+			// the fragmenting
+			maxEventsPerMsg := 20
+			watchRespSent := false
+			for !watchRespSent {
+				if maxEventsPerMsg == 0 {
+					return
+				}
+				for _, fragment := range v3rpc.FragmentWatchResponse(maxEventsPerMsg, wresp) {
```

Comment:
this duplicates the functionality from the server; the proxy can probably passively forward fragments received from the server instead of refragmenting